src/EventListener/DeserializeListener.php

```diff
@@ -32,22 +34,24 @@
  */
 final class DeserializeListener
 {
+    use ContextTrait;
     use ToggleableOperationAttributeTrait;
 
     public const OPERATION_ATTRIBUTE_KEY = 'deserialize';
 
     private $serializer;
-    private $serializerContextBuilder;
+    private $serializerContextFactory;
     private $formats;
     private $formatsProvider;
 
     /**
-     * @param ResourceMetadataFactoryInterface|FormatsProviderInterface|array $resourceMetadataFactory
+     * @param SerializerContextBuilderInterface|SerializerContextFactoryInterface $serializerContextBuilder
+     * @param ResourceMetadataFactoryInterface|FormatsProviderInterface|array     $resourceMetadataFactory
      */
-    public function __construct(SerializerInterface $serializer, SerializerContextBuilderInterface $serializerContextBuilder, $resourceMetadataFactory, ResourceMetadataFactoryInterface $legacyResourceMetadataFactory = null)
+    public function __construct(SerializerInterface $serializer, /* SerializerContextFactoryInterface $serializerContextFactory */$serializerContextBuilder, $resourceMetadataFactory, ResourceMetadataFactoryInterface $legacyResourceMetadataFactory = null)
     {
         $this->serializer = $serializer;
-        $this->serializerContextBuilder = $serializerContextBuilder;
+        $this->serializerContextFactory = $serializerContextBuilder;
```

Comment:
Missing deprecation?

Reply:
Since the service and the method (`createFromRequest`) are deprecated, I didn't thought it was necessary.